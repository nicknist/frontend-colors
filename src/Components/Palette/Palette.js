import React from 'react';

const Palette = ({ name, colors, deletePalette }) => {
  const colorDivs = colors.map(color => {
    const background = {
      margin: '10px',
      padding: '5px',
      width: '10px',
      height: '20px',
      background: `${color}`,
      borderRadius: '1px'
    }
    return (
      <div style={ background }>
      </div>
    )
  })

  return (
    <>
      <h1 className='white'>{name}</h1>
      <button id={name} onClick={() => deletePalette(name)} className='randomizer'>Delete This Palette</button>
      <article className='palette-div'>
        {colorDivs}
      </article>
    </>
  )
}

export default Palette;
