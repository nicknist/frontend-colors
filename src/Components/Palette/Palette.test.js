import React from 'react';
import Palette from './Palette';
import { shallow } from 'enzyme';

describe('Palette', () => {
  it('should match the snapshot', () => {
    const mockColors = [
      {color: 'white', locked: false}
    ]
    const wrapper = shallow(<Palette
      name="waddup"
      colors={mockColors}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot when there is a deletePalette function', () => {
    const mockColors = [
      {color: 'white', locked: false}
    ]
    const wrapper = shallow(<Palette
      name="waddup"
      colors={mockColors}
      deletePalette={jest.fn()}
    />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should run deletePalette when the button is clicked', () => {
    const mockColors = [
      {color: 'white', locked: false}
    ]
    const mockDeletePalette = jest.fn();
    const wrapper = shallow(<Palette
      name="waddup"
      colors={mockColors}
      deletePalette={mockDeletePalette}
    />);

    wrapper.find('button').simulate('click');

    expect(mockDeletePalette).toHaveBeenCalledWith('waddup', undefined, undefined);
  });
})
