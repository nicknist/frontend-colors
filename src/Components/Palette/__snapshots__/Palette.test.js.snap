// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Palette should match snapshot when there is a deletePalette function 1`] = `
<Fragment>
  <h1
    className="white"
  >
    waddup
  </h1>
  <button
    className="randomizer"
    id="waddup"
    onClick={[Function]}
  >
    Delete This Palette
  </button>
  <article
    className="palette-div"
  >
    <div
      key="white"
      style={
        Object {
          "background": "white",
          "borderRadius": "1px",
          "height": "20px",
          "margin": "10px",
          "padding": "5px",
          "width": "10px",
        }
      }
    />
  </article>
</Fragment>
`;

exports[`Palette should match the snapshot 1`] = `
<Fragment>
  <h1
    className="white"
  >
    waddup
  </h1>
  <article
    className="palette-div"
  >
    <div
      key="white"
      style={
        Object {
          "background": "white",
          "borderRadius": "1px",
          "height": "20px",
          "margin": "10px",
          "padding": "5px",
          "width": "10px",
        }
      }
    />
  </article>
</Fragment>
`;
